2022.2: Use new profile stubber

knatraj-rythmos opened the pull request:
This profile stubber will correctly stub CustomMarshalInfo attributes.

Parent: https://jira.unity3d.com/browse/UUM-21101
Backport: https://jira.unity3d.com/browse/UUM-21132

- Should this pull request have release notes?
  - [x] Yes
  - [ ] No
- Do these changes need to be back ported?
  - [x] Yes
  - [ ] No
- Do these changes need to be upstreamed to [mono/mono](https://github.com/mono/mono) or [dotnet/runtime](https://github.com/dotnet/runtime) repositories?
  - [ ] Yes
  - [x] No

Reviewers: please consider these questions as well! :heart:

**Release notes**

Fixed UUM-21101 @knatraj-rythmos:
Mono: Correct the CustomMarshal attribute definition for the Control::Font property in System.Windows.Forms.dll in the AOT profile.

Comment to Reviewers: 
Backport is a [CleanGraft]
Related IL2CPP PR Link: https://github.cds.internal.unity3d.com/unity/il2cpp/pull/4945

Trunk PR: https://github.com/Unity-Technologies/mono/pull/1714

joshpeterson commented:
Let's hold off on merging this for a little while. It might have caused a problem on the unity-main branch. I'm investigating now, and should have an answer in the next day or so.

knatraj-rythmos commented:
> Let's hold off on merging this for a little while. It might have caused a problem on the unity-main branch. I'm investigating now, and should have an answer in the next day or so.

Sure lets wait until you finish investigating it. Thanks.

Commit:
Avoid stubbing CustomMarshalInfo (UUM-21101)

This profile stubber will avoid stubbing CustomMarshalIntfo at all.

In a previous PR:

https://github.com/Unity-Technologies/mono/pull/1711

we attempted to properly stub the CustomMarshalInfo, but we found it was
better to simply avoid stubbing it.

joshpeterson commented:
@knatraj-rythmos: Please use the change from https://github.com/Unity-Technologies/mono/pull/1714 instead. Thanks!

knatraj-rythmos commented:
> @knatraj-rythmos: Please use the change from #1714 instead. Thanks!

Sure I will go ahead and do that changes. Thanks.
